src/app/test/db/mongo/models_test.py

```diff
@@ -612,3 +645,35 @@ def test_only_one_local_garden_may_exist(self, testgarden):
         should raise an exception"""
         with pytest.raises(NotUniqueError):
             Garden(name=f"not{testgarden.name}", connection_type="LOCAL").save()
+
+    def test_child_garden_system_update(self, child_garden, child_system_v2):
```

Comment:
This test is a good one to have, but should we also have one that tests that the update works when a new system with a matching "triple" shows up?  That's the case that inspired these changes in the first place.